reference/proto-design-guidelines.md

```diff
@@ -0,0 +1,115 @@
+
+#### "enablePhysics"
+
+Because sometimes, we want to simulate the physical motion of an object and sometimes we want to save computation time by not simulating the physics of the object, it is good also to add a `enablePhysics` boolean field that allows the user to enable or disable physics simulation for an object.
+For example, a box may be meant to be moveable by a robot or may be glued to the floor or so heavy that it won't move.
+In that case, it is nice to be able to specify if we want to simulate the physical motion of that object or not.
+Not simulating the physics saves computation time and increase the stability of the simulation.
+However, some objects are not meant to move during a simulation.
+This includes buildings, traffic signs, trees, mailboxes, etc.
+Therefore such objects should not contain a `Physics` node and obviously should not expose a `enablePhysics` field.
+
+#### Other Fields
+
+In PROTO nodes that correspond to real world objects, it is not recommend to add fields such as `mass` or `scale`.
+This is because in real life, you cannot change the mass or the size of an object.
+And we don't want to encourage users of the PROTO nodes to do so, as it may lead to non-realistic models and numerical instabilities.
+Some exceptions however exist.
+For example, it may be convenient to add a `size` field to a Matryoshka doll PROTO.
+Nevertheless, such a `size` field should be limited to a minimal and a maximal size or to a limited range of integer values.
+Therefore it could be either a floating point or an integer value.
+
+For some PROTO nodes, it could be useful to expose a `color` field.
+For example, a car or a color pencil PROTO could have a `color` field exposed.
+However, that should be limited to objects that are available in different colors.
+For example, it should not be used for a fire hydrant (usually always red) or a fork (usually always metallic grey).
+The `color` field may be specified as a `Color` node if any color is available or as a string (if only a limited number of colors is available for that object).
+
+Depending on the object, a number of texture fields may be useful.
+For example a painting PROTO could have the painting specified in a `picture` field and the frame texture specified in a `frame` field.
+
+Generally, the number of exposed fields should be minimal in order to guarantee that the resulting object will be realistic and correspond to the original idea of the PROTO.
+Also, floating point and integer values should be generally constrained between a minimum and a maximum value to ensure a realistic and stable result.
+
+### Example
+
+Here is a simple example of a good PROTO declaration (the implementation is not shown):
+
+```
+#VRML_SIM R2018b utf8
+# license: Creative Commons Attribution 4.0 International License
+# tags: static
```

Comment:
It is not very important but currently in our PROTO and in our other PROTO example doc we put 'tags' and 'license' in the other order.